src/strings.js

```diff
@@ -40,54 +40,115 @@ function rewritePatternForDotAll(pattern) {
   return cachedRegExp[pattern];
 }
 
-engine.indexOf = function(coll, substr){
+engine.indexOf = function (coll, substr) {
   const str = misc.singleton(coll, 'String');
   return util.isEmpty(substr) || util.isEmpty(str) ? [] : str.indexOf(substr);
 };
 
-engine.substring = function(coll, start, length){
+engine.substring = function (coll, start, length) {
   const str = misc.singleton(coll, 'String');
   if (util.isEmpty(str) || util.isEmpty(start) || start < 0 || start >= str.length) {
-    return  [];
+    return [];
   }
   if (length === undefined || util.isEmpty(length)) {
     return str.substring(start);
   }
   return str.substring(start, start + length);
 };
 
-engine.startsWith = function(coll, prefix){
+engine.startsWith = function (coll, prefix) {
   const str = misc.singleton(coll, 'String');
   return util.isEmpty(prefix) || util.isEmpty(str) ? [] : str.startsWith(prefix);
 };
 
-engine.endsWith = function(coll, postfix) {
+engine.endsWith = function (coll, postfix) {
   const str = misc.singleton(coll, 'String');
   return util.isEmpty(postfix) || util.isEmpty(str) ? [] : str.endsWith(postfix);
 };
 
-engine.containsFn = function(coll, substr){
+engine.containsFn = function (coll, substr) {
   const str = misc.singleton(coll, 'String');
   return util.isEmpty(substr) || util.isEmpty(str) ? [] : str.includes(substr);
 };
 
-engine.upper = function(coll){
+engine.upper = function (coll) {
   const str = misc.singleton(coll, 'String');
   return util.isEmpty(str) ? [] : str.toUpperCase();
 };
 
-
-engine.lower = function(coll){
+engine.lower = function (coll) {
   const str = misc.singleton(coll, 'String');
   return util.isEmpty(str) ? [] : str.toLowerCase();
 };
 
+engine.joinFn = function (coll, separator) {
+  const stringValues = coll.map((n) => {
+    const d = util.valData(n);
+    if (typeof d === "string") {
+      return d;
+    }
+    return null;
+  });
+  return stringValues.join(separator);
+};
+
+engine.splitFn = function (coll, separator) {
+  const strToSplit = misc.singleton(coll, 'String');
+  return util.isEmpty(strToSplit) ? [] : strToSplit.split(separator);
+};
+
+engine.trimFn = function (coll) {
+  const stringValues = coll.map((n) => {
+    const d = util.valData(n);
+    if (typeof d === "string") {
+      return d.trim();
+    }
+    return null;
```

Comment:
This will put a null in the output array.  I think it should just be skipped (which would require changing "map" to a loop).

Reply:
Not sure if this is important, but If we skip null, then `split` after `join` with the same delimiter will result in an array with another length. I guess we can replace `null` with an empty string, or maybe convert 'd' to a string.
If my assumption is correct, then it may be good to add a unit test to test `split` after `join`.

Reply:
I'll take this up with the fhirpath definition...